lib/web_ui/lib/src/engine/safe_browser_api.dart

```diff
@@ -814,12 +814,14 @@ class GlContext {
 
   /// Returns image data in a form that can be used to create Canvas
   /// context patterns.
-  Object? readPatternData() {
+  Object? readPatternData(bool isOpaque) {
     // When using OffscreenCanvas and transferToImageBitmap is supported by
     // browser create ImageBitmap otherwise use more expensive canvas
-    // allocation.
+    // allocation. However, transferToImageBitmap does not properly preserve
+    // the alpha channel, so only use it if the pattern is opaque.
```

Comment:
I wonder if this is because on the following line we should also specify the `alpha` parameter:

https://github.com/flutter/engine/blob/ba8f1c5a7183ebdc2729aebbb66815f87dc21075/lib/web_ui/lib/src/engine/safe_browser_api.dart#L434

According to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#:~:text=WebGL%20context%20attributes%3A-,alpha,-%3A%20Boolean%20that%20indicates): "alpha: Boolean that indicates if the canvas contains an alpha channel. If set to false, the browser now knows that the backdrop is always opaque, which can speed up drawing of transparent content and images."

Reply:
I actually tried this already! It did nothing unfortunately. I believe alpha is true by default.